### DISCUSSION TITLE
Interpreting Geometric Adstock

### DISCUSSION THREAD

**Discussion by bstuddard** [pymc-labs/pymc-marketing#280] [Q&A]
I'm looking for a way to interpret the geometric adstock results (alpha parameter) to translate it into an approximate number of periods before decay/effect is negligible, would this be an appropriate approach to back into that interpretation (assume alpha is the mean posterior coefficient)?

```python
import math

alpha = 0.2

decay_threshold = 0.01 # ex: determine when ad effect is decayed to 1%

num_periods = math.log(decay_threshold, alpha)

print(f'It takes approximately {round(num_periods,2)} periods to reach {decay_threshold:.2%} of the original channel effect.')
```

**Comment by juanitorduz** [comment]
I do not have a concrete answer but here are some ideas. First of all note that the carryover parameter implicitly depends on the maximum number of period to consider (see `l_max` in https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/mmm/transformers.py#L82). I guess you could compute the desired quantity using the explicit adstock effect formula as a heuristic approximation. Note that we are normalizing after the transformation (https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/mmm/delayed_saturated_mmm.py#L121) , so the overall contribution after the transformation should match the input spend.
